internals/overlord/logstate/gatherer_test.go

```diff
@@ -218,6 +219,92 @@ func (s *gathererSuite) TestRetryLoki(c *C) {
 	}
 }
 
+// Test to catch race conditions in gatherer
+func (s *gathererSuite) TestRace(c *C) {
```

Comment:
Can we make this name more specific, like `TestXyzRace` with what it's actually testing a race in or for? And after that let's remove the redundant doc comment.

Reply:
This is a pretty non-specific test, it's just designed to hammer the log gatherer and find any potential race conditions. It's now finding some artificial race conditions though. Needs a rewrite.

Reply:
I rewrote the test so it passes. I'm still not sure what to call it though - do you have any suggestions?